localScrapper.js

```diff
@@ -16,8 +27,15 @@ function getFileInfos(fileName, dirPath, stat) {
   let languageString
   let extension
 
-  const lastIndexOfSlash = dirPath.lastIndexOf('/')
-  const dirName = lastIndexOfSlash !== -1 ? dirPath.substr(lastIndexOfSlash + 1) : '.'
+  const directories = dirPath.split('/')
+  // Group by subdirectories : Take the name of the last directory (/karaoke/TEMP/Anime >> Anime)
+  // Group by top directory : Take the name of the topmost directory, excluding the base one :
+  //   /karaoke/TEMP/Anime >> TEMP
+  //   if anything is in the top directory, it will be indexed as such.
+  const dirName = groupBySubdirectories ?
+      directories[directories.length - 1] :
+      directories[1] || directories[0]
+
```

Comment:
We won't use the directories after that so we can just use pop >> Yes, but I feel bad about uselessly modifying this array. Compromise : `_.last(directories)` ? (I haven't checked lodash's api but I'm pretty sure this exists)

I don't understand the directories[1] || directories[0], is it to index the karaoke directory in itself? >> yep, just in case.